sktime/transformations/series/multivariate_compositor.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__all__ = ["MultivariateCompositor"]
+__author__ = ["Svea Meyer"]
+
+import pandas as pd
+from sklearn.base import clone
+from sktime.transformations.base import _SeriesToSeriesTransformer
+from sktime.utils.validation.series import check_series
+
+
+class MultivariateCompositor(_SeriesToSeriesTransformer):
+    """
+    Parameters
+    ----------
+    transformer : series-to-series transformer to be applied to each series
+    columns : names of columns that are supposed to be transformed
+    """
+
+    _required_parameters = ["transformer"]
+    _tags = {
+        "multivariate-only": True,
+    }
+
+    def __init__(self, transformer, columns="all"):
+        self.transformer = transformer
+        self.columns = columns
+        self.transformers_ = None
+        super(MultivariateCompositor, self).__init__()
+
+    def fit(self, Z, X=None):
+        """
+        Iterates over columns (series) and applies the fit function of the transformer
+        """
+        if not isinstance(Z, pd.DataFrame):
+            raise ValueError("Z needs to be a multivariate Pandas Series")
+        self._is_fitted = False
+        z = check_series(Z)
+        if self.columns == "all":
+            self.columns = z.columns
+
+        # make sure z contains all columns that the user wants to transform
+        Z_wanted_keys = set(self.columns)
+        Z_new_keys = set(z.columns)
+        difference = Z_wanted_keys.difference(Z_new_keys)
+        if len(difference) != 0:
+            raise ValueError("Missing columns" + str(difference) + "in Z.")
+
+        self.transformers_ = {}
+        for colname in self.columns:
+            transformer = clone(self.transformer)
+            self.transformers_[colname] = transformer
+            self.transformers_[colname].fit(z[colname], X)
+            # self.transformers_[colname].is_fitted = True
+        self._is_fitted = True
+        return self
+
+    def transform(self, Z, X=None):
+        """
+        calls transform on every single transformer (one transformer per series)
+        """
+        if not isinstance(Z, pd.DataFrame):
+            raise ValueError("Z needs to be a multivariate Pandas Series")
```

Comment:
this should be included in `check_series()` function with a new param `enforce_multivariate`. Also above you have this check which can be replaced then as well with simply z = check_series(Z, enforce_multivariate=True).
Also the error should rather be `"Z needs to be a pd.DataFrame"`

Reply:
Agreed, this might still change depending on what we decided about the question of whether it's ok to have a multivariate-only tag after all (see discussion above)

Reply:
I think its not dependent to the tags, but rather assessing the input? Tags are mainly for tests

Reply:
I wasn't aware of that, thanks @aiwalter! So far `check_series` however, doesn't have an argument `enforce_multivariate` (only enforce_univariate) so I would need to introduce that. And I think that would only make sense if we end up having the tag `multivariate-only`. Otherwise, this function would also have to accept pd.Series.

Reply:
i think to add `enforce_multivariate` to `check_series` is fine, would be great if you can do it with this PR.?

Reply:
sure, will do :)

Reply:
its not dependent to any tag discussion, bcz having it will help to replace your code here and in other places where we want to do `check_series(Z, enforce_multivariate=True)` in the future.